src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -403,6 +404,17 @@ void WriteId(StreamWriter writer, string input)
             }
         }
 
+        private async Task WriteToRequestBodyStreamAsync(ArraySegment<byte> buffer)
+        {
+            await _requestBodyStream.WriteAsync(buffer.Array, buffer.Offset, buffer.Count, _token).ConfigureAwait(false);
+
+            if (_isInitialWrite)
+            {
+                _isInitialWrite = false;
+                await _requestBodyStream.FlushAsync(_token).ConfigureAwait(false);
```

Comment:
This probably forces us to actually send the data over the wire initially, and avoid holding the connection open (but not used).